lib/elixir/lib/code.ex

```diff
@@ -1095,7 +1099,16 @@ defmodule Code do
   defp do_fetch_docs(bin_or_path) do
     case :beam_lib.chunks(bin_or_path, [@docs_chunk]) do
       {:ok, {_module, [{@docs_chunk, bin}]}} ->
-        :erlang.binary_to_term(bin)
+        case :erlang.binary_to_term(bin) do
+          {:docs_v1, _, _, _, _, _, _} = chunk ->
+            chunk
+
+          tuple when is_tuple(tuple) and is_atom(elem(tuple, 0)) ->
```

Comment:
@josevalim I see, you want to make `fetch_docs` transparent and put the burden of knowing what's in the chunks entirely on the caller.

With this change and the other about rescuing `binary_to_term` I'm thinking that we should also revisit the typespec for `Code.fetch_docs/1`. We can't really promise to return the current spec unless we match on the result of `binary_to_term`. What should be the approach for the spec in this case? I would like to make the spec more useful than simply `@spec fetch_docs(module) :: term | {:error, term}`.

Reply:
We can try:

    @spec fetch_docs(module) :: {:docs_v1, ...} | {:error, term} | future_formats when future_formats: term

But this may have negative implications in dialyzer. We can give it a try though.